app/src/main/java/com/ciandt/dragonfly/example/shared/BaseActivity.kt

```diff
@@ -114,6 +123,17 @@ abstract class BaseActivity(protected var hasDebugDrawer: Boolean = true) : AppC
         return pendingPermissions
     }
 
+    private fun checkLogin() {
+        if (!isUserLogged() && this !is LoginActivity) {
+            startActivity(LoginActivity.create(this))
```

Comment:
Far as I could see, LoginActivity extends BaseActivity. In this case, we would create a circular dependency between these two classes.  I suggest to abstract the call to LoginActivity inside our IntentHelper (IntentHelper would call LoginActivity.newIntent() - or LoginActivity.create() )